config/v1/types_apiserver.go

```diff
@@ -30,6 +30,15 @@ type APIServerSpec struct {
 	// - ConfigMap.Data["ca-bundle.crt"] - CA bundle.
 	// +optional
 	ClientCA ConfigMapNameReference `json:"clientCA"`
+	// extraCORSAllowedOrigins is a field of allowed CORS that should be added on top of the basic CORS needed by the
+	// infrastructure in order to allow in-cluster components to access the API and OAuth servers using Javascript.
+	// The field values should follow the syntax of the corsAllowedOrigins field.
+	// If any operator means to use the field, it should only be checking that the value that it requires to appear
+	// is/is not present, and add it if it isn’t.
+	// We can assume that the array of the values of the field is managed by other operators, so the operator managing
+	// a certain value should remove it in case the operator itself is being deleted. This removal must happen before
+	// the resource that is represented by the value being removed (e.g. hostname from a route status) is itself removed.
+	ExtraCORSAllowedOrigins []string `json:extraCORSAllowedOrigins`
```

Comment:
I think I would prefer ("extra" seems like internal detail):

```go
CORSAllowedOrigins []string `json:corsAllowedOrigins`
```

Reply:
> I think I would prefer ("extra" seems like internal detail):
> 
> ```go
> CORSAllowedOrigins []string `json:corsAllowedOrigins`
> ```

I'm fine the name suggested here, but the doc should indicate that the list isn't fully authoritative

Reply:
<bike-shedding>"extra" is more precise, without it is misleading. Let's use the former.</bike-shedding>

Reply:
Also add the json tag, make it `+optional` and add `omitempty` to the json tag.